gensim/corpora/mmcorpus.py

```diff
@@ -19,21 +19,78 @@
 logger = logging.getLogger('gensim.corpora.mmcorpus')
 
 
-class MmCorpus(matutils.MmReader, IndexedCorpus):
+try:
```

Comment:
Can you move this try/except section to `matutils.py` and restore all imports (this is simplest and clearest way)

Reply:
I tried moving the import to matutils, but it causes some issues with circular imports.

For example, if I put the following in matutils.py:

```python
try:
    # try loading cython version of mmreader
    from gensim.corpora._mmreader import MmReader
except ImportError:
    # else fallback to python version
    class MmReader(object):
        pass
```

When the try block tries to import gensim.corpora._mmreader, that executes gensim.corpora.\_\_init\_\_, which tries to import gensim.corpora.mmcorpus.MmCorpus, which unfortunately tries to import matutils.MmReader, which isn't defined yet (as we are still in the try block).

You end up with a exception like:

```python
----> 1 import gensim

~/Projects/gensim/gensim-mmreader/gensim/__init__.py in <module>()
      4 """
      5
----> 6 from gensim import parsing, matutils, interfaces, corpora, models, similarities, summarization, utils  # noqa:F401
      7 import logging
      8

~/Projects/gensim/gensim-mmreader/gensim/matutils.py in <module>()
     33 try:
     34     # try loading cython version of mmreader
---> 35     from gensim.corpora._mmreader import MmReader
   
~/Projects/gensim/gensim-mmreader/gensim/corpora/__init__.py in <module>()
      6 from .indexedcorpus import IndexedCorpus  # noqa:F401 must appear before the other classes
      7
----> 8 from .mmcorpus import MmCorpus  # noqa:F401
      9 from .bleicorpus import BleiCorpus  # noqa:F401
   
~/Projects/gensim/gensim-mmreader/gensim/corpora/mmcorpus.py in <module>()
     21
---> 22 class MmCorpus(matutils.MmReader, IndexedCorpus):
     23     """
   
AttributeError: module 'gensim.matutils' has no attribute 'MmReader'
```

I couldn't think of an easy way around this, short of moving _mmreader out of the corpora package, but I'll keep playing around with it.

Reply:
Oh, that's really sad, I understand what happens :( To avoid this, we need to move `_mmreader` from corpora (but I have no idea what's better place for it).